src/sp/it/pl/util/system/Os.kt

```diff
@@ -45,20 +45,22 @@ enum class Os {
     companion object {
 
         /** @return the current operating system */
-        @JvmStatic val current: Os
-            get() {
-                val osName = System.getProperty("os.name")
-                if (osName.startsWith("Windows")) return WINDOWS
-                if (osName.startsWith("Mac")) return OSX
-                if (osName.startsWith("SunOS")) return UNIX
-                if (osName.startsWith("Linux")) {
-                    val javafxPlatform = System.getProperty("javafx.platform")
-                    if (!("android"==javafxPlatform || "Dalvik"==System.getProperty("java.vm.name")))  // if not Android
-                        return UNIX
-                }
-                return UNKNOWN
+        @JvmStatic val current: Os = run {
+            val osName = System.getProperty("os.name")
+            when {
+                osName.indexOf("win")!=-1 -> WINDOWS
+                osName.indexOf("mac")!=-1 -> OSX
+                // various forms of unix
+                osName.startsWith("SunOS") || osName.indexOf("nix")!=-1 || osName.indexOf("freebsd")!=-1
```

Comment:
Cant this be rewritten to commas as multiple when predicates of the same when clause? Maces mor esense that way to me.

Reply:
oh, right, didn't think of that

Reply:
```
osName.startsWith("SunOS") -> UNIX
osName.indexOf("nix")>-1  -> UNIX
osName.indexOf("freebsd")>-1 -> UNIX
```
Like this?

Reply:
More like this:
```
osName.startsWith("SunOS"),
osName.indexOf("nix")>-1,
osName.indexOf("freebsd")>-1 -> UNIX
```
If no, the version in your last reply is stil superior over whats commited I think.

Reply:
IDEA: "Deprecated syntax. Use '||' instead of commas in when-condition for 'when' without argument"

I guess that's that.

Reply:
Nooooooooooooooo. That seems weird.
DO you not think
```
osName.startsWith("SunOS") -> UNIX
osName.indexOf("nix")>-1  -> UNIX
osName.indexOf("freebsd")>-1 -> UNIX
```
is still better?

Reply:
I think `||` is fine, since it's just a few different names for unix distributions. But it doesn't really matter to me. I can understand that `,`

Reply:
Comma is not unusual syntax, consider enum constants in a when expression. Comma is depcrecated only for when without an argument. I searched, but was unable to find out why.

Anyway, the comma syntax is nice because it 'joins' the branches into one (not necessarily all branches that return same value!), which is helpful to emphasize intent and similarity. But they still remain separate branches.

That is not the case when || is used, which, while not really a problem, does in my opinion detract from the homogeneity of the entire when block - suddenly some branches are merged into one logical operation. That is not good. It is foolish to assume reader can always recognize when the logical expression is an aggregate of multiple branches and when not. This should be clearly expressed.

Another minor benefit of the comma or separate branch solutions is that the code is vertical, not horizontal. Yes it is more lines but:
- all branches are readily visible
- all branches are equivalent
- its better for the eyes
- its better for git (clear what changed, avoids conflicts)
- avoids long lines (and wrapping issues)
- avoids dense blobs of code

There really is no good reason to use || in case like this, when it is semantically incorrect.

Reply:
You could use || and still align it vertically.